parsers/nmparser/get_model_output.go

```diff
@@ -67,48 +68,137 @@ func GetModelOutput(filePath string, verbose bool, noExt bool, noGrd bool, noCov
+		for n := nameCount; n < omegaCount; n++ {
+			results.ParametersData[i].RandomEffectSDSE.Omega = append(results.ParametersData[i].RandomEffectSDSE.Omega, 0)
+		}
+		nameCount = len(results.ParametersData[i].RandomEffectSDSE.Sigma)
+		for n := nameCount; n < sigmaCount; n++ {
+			results.ParametersData[i].RandomEffectSDSE.Sigma = append(results.ParametersData[i].RandomEffectSDSE.Sigma, 0)
+		}
+
+		// parameter names
+		nameCount = len(results.ParameterNames.Theta)
+		for n := nameCount; n < thetaCount; n++ {
+			results.ParameterNames.Theta = append(results.ParameterNames.Theta, fmt.Sprintf("Theta%v", n+1))
+		}
+		nameCount = len(results.ParameterNames.Omega)
+		for n := nameCount; n < omegaCount; n++ {
+			results.ParameterNames.Omega = append(results.ParameterNames.Omega, fmt.Sprintf("Omega%v", n+1))
+		}
+		nameCount = len(results.ParameterNames.Sigma)
+		for n := nameCount; n < sigmaCount; n++ {
+			results.ParameterNames.Sigma = append(results.ParameterNames.Sigma, fmt.Sprintf("Sigma%v", n+1))
+		}
+
 	}
 
-	for n := 0; n < len(results.ShrinkageDetails); n++ {
-		if len(results.ShrinkageDetails[n].Eta.SD) == 0 {
-			dim := 0
-			for i := range results.ParameterStructures.Omega {
-				if results.ParameterStructures.Omega[i] > 0 {
-					dim++
-				}
+	// shrinkage eta's
+	if etaCount > 0 {
+		for n := 0; n < len(results.ShrinkageDetails); n++ {
+			if len(results.ShrinkageDetails[n].EtaSD) == 0 {
+				results.ShrinkageDetails[n].EtaSD = make([]float64, etaCount)
 			}
-			if dim > 0 {
-				results.ShrinkageDetails[n].Eta.SD = make([]float64, dim)
-				results.ShrinkageDetails[n].Eta.VR = make([]float64, dim)
-				// Ebv follows Eta
-				results.ShrinkageDetails[n].Ebv.SD = make([]float64, dim)
-				results.ShrinkageDetails[n].Ebv.VR = make([]float64, dim)
+			if len(results.ShrinkageDetails[n].EtaVR) == 0 {
+				results.ShrinkageDetails[n].EtaVR = make([]float64, etaCount)
+			}
+			if len(results.ShrinkageDetails[n].EbvSD) == 0 {
+				results.ShrinkageDetails[n].EbvSD = make([]float64, etaCount)
+			}
+			if len(results.ShrinkageDetails[n].EbvVR) == 0 {
+				results.ShrinkageDetails[n].EbvVR = make([]float64, etaCount)
+			}
+			if len(results.ShrinkageDetails[n].EtaBar) == 0 {
+				results.ShrinkageDetails[n].EtaBar = make([]float64, etaCount)
+			}
+			if len(results.ShrinkageDetails[n].EtaBarSE) == 0 {
+				results.ShrinkageDetails[n].EtaBarSE = make([]float64, etaCount)
+			}
+			if len(results.ShrinkageDetails[n].Pval) == 0 {
+				results.ShrinkageDetails[n].Pval = make([]float64, etaCount)
+			}
+			if len(results.ShrinkageDetails[n].NumSubjects) == 0 {
+				results.ShrinkageDetails[n].NumSubjects = make([]float64, etaCount)
 			}
 		}
 
-		if len(results.ShrinkageDetails[n].Eps.SD) == 0 {
-			dim := 0
-			for i := range results.ParameterStructures.Sigma {
-				if results.ParameterStructures.Sigma[i] > 0 {
-					dim++
+		// shrinkage eps's
+		if epsCount > 0 {
+			for n := 0; n < len(results.ShrinkageDetails); n++ {
```

Comment:
range

Reply:
Go uses a copy of the value instead of the value itself within a range clause, so the array is lost when exiting the scope of the range.